doc/configuration.rst

```diff
@@ -428,40 +431,38 @@ You can create a custom sort key callable for the following configurations:
 * :ref:`subsection_order <sub_gallery_order>`
 * :ref:`minigallery_sort_order <minigallery_order>`
 
-When creating a custom sort key callable, you must ensure that the ``__repr__``
-is stable across runs. Sphinx determines if the build environment has
-changed, and thus if *all* documents should be rewritten, by examining the
-config values using ``md5(str(obj).encode()).hexdigest()`` in
-``sphinx/builders/html.py``. Default class instances in Python have their
-memory address in their ``__repr__`` which is why generally the ``__repr__``
-changes in each build.
+The best way to do this is to make use of a single instance of each class you want
+to use in some fully resolvable path, and then tell sphinx-gallery to resolve the
+callable from the fully qualified name. For example, you could write a function
+``def my_sorter`` and put it in ``mymod/utils.py``, then use::
 
-For example, in :class:`sphinx_gallery.sorting.ExplicitOrder` stability is
-ensured via the custom ``__repr__``::
-
-    def __repr__(self):
-        return '<%s: %s>' % (self.__class__.__name__, self.ordered_list)
+    sphinx_gallery_conf = {
+    #...,
+    "minigallery_sort_order": "mymod.utils.my_sorter",
+    #...
+    }
 
-Therefore, the files are only all rebuilt when the specified ordered list
-is changed.
+Another option is to add documentation-specific stuff in a different path and ensure
+that it can be resolved at build time. For example, you could create a file
+``doc/sphinxext.py``::
 
-We recommend that you use the :class:`sphinx_gallery.sorting.FunctionSortKey`
-because it will ensure that the ``__repr__`` is stable across runs.
+    def plotted_sorter(fname):
+        return not fname.startswith("plot_"), fname
+    )
 
-Create your sort key callable by instantiating a
-:class:`~sphinx_gallery.sorting.FunctionSortKey` instance with your sort key
-function. For example, the following ``minigallery_sort_order`` configuration
-(which sorts on paths) will sort using the first 10 letters of each filename:
+And set in your configuration::
 
-.. code-block:: python
+    sys.path.insert(0, os.path.dirname(__file__))
 
     sphinx_gallery_conf = {
     #...,
-    "minigallery_sort_order": FunctionSortKey(
-        lambda filename: filename[:10]),
+    "minigallery_sort_order": "sphinxext.plotted_sorter",
     #...
     }
 
+And sphinx-gallery would resolve ``"sphinxext.function_sorter"`` to the
+``function_sorter`` object because the ``doc/`` directory is first on the path.
```

Comment:
```suggestion
And sphinx-gallery would resolve ``"sphinxext.plotted_sorter"`` to the
``plotted_sorter`` object because the ``doc/`` directory is first on the path.
```